File: passenger-service/src/main/java/com/dulfinne/taxi/passengerservice/dto/response/PassengerResponse.java
package com.dulfinne.taxi.passengerservice.dto.response;

import com.dulfinne.taxi.passengerservice.model.Payment;
import lombok.Builder;

@Builder
public record PassengerResponse(
    Long id,
    String username,
    String firstName,
    String lastName,
    String phoneNumber,
    Payment payment,
    Integer rideCount,
    Double averageRating
) {
}

File: ride-service/src/main/kotlin/com/dulfinne/taxi/rideservice/client/dto/PassengerResponse.kt
package com.dulfinne.taxi.rideservice.client.dto

import com.dulfinne.taxi.rideservice.model.Payment

data class PassengerResponse(
    val id: Long,
    val username: String,
    val firstName: String,
    val lastName: String,
    val phoneNumber: String,
    val payment: Payment,
    val rideCount: Int,
    val averageRating: Double
)
